packages/testing-lib/src/RTLOverrides.ts

```diff
@@ -0,0 +1,31 @@
+import * as RTL from '@testing-library/react';
+
+type Exists<X, Y extends keyof X | string, Fallback = any> = Y extends keyof X
+  ? X[Y]
+  : Fallback;
+
+/**
+ * Re-exports `renderHook` from `"@testing-library/react"` if it exists,
+ * or from `"@testing-library/react-hooks"`
+ *
+ * (used when running in a React 17 test environment)
+ */
+export const renderHook: Exists<typeof RTL, 'renderHook'> =
+  (RTL as any).renderHook ??
```

Comment:
Does `renderHook` only get exported from certain versions of RTL or something? What's the uncertainty that makes use need the `Exist` type?

Reply:
Yeah. Prior to `@testing-library/react@13.1.0`, `renderHook` was only available in the standalone, `/react-hooks` package.  The `react-hooks` package has since been soft-deprecated for React 18 use-cases, but `/react > 13` is not compatible with React 18, so we need to toggle which version of `renderHook` we import